recipes/xkbcommon/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class XkbcommonConan(ConanFile):
+    name = "xkbcommon"
+    description = "keymap handling library for toolkits and window systems"
+    topics = ("conan", "xkbcommon", "keyboard")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/xkbcommon/libxkbcommon"
+    license = "MIT"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_x11": [True, False],
+        "with_wayland": [True, False],
+        "docs": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_x11": True,
+        "with_wayland": False,
+        "docs": False
+    }
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("This library is only compatible with Linux")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def _system_package_architecture(self):
+        if not tools.cross_building(self.settings):
+            return ""
+
+        if tools.os_info.with_apt:
+            if self.settings.arch == "x86":
+                return ":i386"
+            elif self.settings.arch == "x86_64":
+                return ":amd64"
+            elif self.settings.arch == "armv6" or self.settings.arch == "armv7":
+                return ":armel"
+            elif self.settings.arch == "armv7hf":
+                return ":armhf"
+            elif self.settings.arch == "armv8":
+                return ":arm64"
+
+        if tools.os_info.with_yum:
+            if self.settings.arch == "x86":
+                return ".i686"
+            elif self.settings.arch == "x86_64":
+                return ".x86_64"
+        return ""
+
+    def _system_package_name(self, package):
+        return package + self._system_package_architecture()
+
+    def system_requirements(self):
+        pack_names = []
+        if tools.os_info.with_apt:
+            pack_names.append(self._system_package_name("xkb-data"))
+            if self.options.with_x11:
+                pack_names.extend([self._system_package_name("libxcb-xkb-dev"),
+                                   self._system_package_name("libxcb1-dev")])
+        elif tools.os_info.with_yum:
+            pack_names.append(self._system_package_name("xkeyboard-config"))
+            if self.options.with_x11:
+                pack_names.append(self._system_package_name("libxcb-devel"))
+
+        if pack_names:
+            installer = tools.SystemPackageTool()
+            for item in pack_names:
+                installer.install(item)
```

Comment:
system_requirements again (like with SDL2)

I would really like to hold back with those right now

See also https://github.com/conan-io/hooks/issues/100